### PR TITLE
Implemented code sharing and fixed packaging

### PR DESCRIPTION
**Pull request by esamattis**
I've added an implementation of client side function sharing. I don't know what was your vision of this, but at least consider this as encouragement for implementing it :) I think that this is a really awesome project!

Also I had to fix this https://github.com/mauricemach/zappa/issues/#issue/27/comment/802075

### CODE REVIEW

**Comment by esamattis** [comment]
I've implemented even better code sharing features directly for Express. It uses UglifyJS as minifier and prettyfier.

https://github.com/epeli/reallyexpress

These might be usable with Zappa too.

File: lib/zappa.coffee [hunk 1]
```diff
@@ -100,6 +102,17 @@ class App
     @socket_handlers = {}
     @msg_handlers = {}
 
+
+    @shared_functions = {}
+    @_shared_code = ""
+    @_shared_code_dirty = false
+
+    @http_server.get "/shared.js", (req, res) =>
+      res.contentType 'bla.js'
```
Comment (defunkt) [inline lib/zappa.coffee:111]: Is `bla.js` intentional?

Comment (esamattis) [inline lib/zappa.coffee:111]: I think not :P

Comment (mauricemach) [inline lib/zappa.coffee:111]: Intentional! :)

`res.contentType` previously only accepted a filename as param, even though only the extension was used.

**Comment by mauricemach** [comment]
Thanks for the encouragement, and sorry for taking so long!

In 0.2.0beta you have `shared` blocks. They're run on the server and available to the client too.

```
shared '/index.js', ->
  def sum: (a, b) -> a + b

  helper role: (name) ->
    unless @user.role is name
      if request? then redirect '/login'
      else if window? then alert "This is not the page you're looking for."
      else if socket? then client.disconnect()

get '/admin': ->
  role 'admin'
  # admin stuff

at 'delete everything': ->
  role 'admin'

client '/index.js': ->
  get '#/admin': ->
    role 'admin'
    # admin stuff
```

**Pull request closed without merging**